docs/core/extensions/timer-service.md

```diff
@@ -0,0 +1,90 @@
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> - Implement the <xref:Microsoft.Extensions.Hosting.IHostedService>, and <xref:System.IAsyncDisposable> interfaces.
+> - Create a timer-based service.
+> - Register the custom implemenation with dependency injection and logging.
+
+## Prerequisites
+
+- The [.NET 5.0 SDK or later](https://dotnet.microsoft.com/download/dotnet)
+- A .NET integrated development environment (IDE)
+  - Feel free to use [Visual Studio](https://visualstudio.microsoft.com)
+
+## Create timer service
+
+The timer-based background service makes use of the <xref:System.Threading.Timer?displayProperty=fullName> class. The timer triggers the `DoWork` method. The timer is disabled on <xref:Microsoft.Extensions.Hosting.IHostLifetime.StopAsync(System.Threading.CancellationToken)?displayProperty=nameWithType> and disposed when the service container is disposed on <xref:System.IAsyncDisposable.DisposeAsync?displayProperty=nameWithType>:
+
+Replace the contents of the `Worker` from the template with the following C# code, and rename the file to *TimerService.cs*:
+
+:::code source="snippets/workers/timer-service/TimerService.cs" highlight="40,47-50":::
+
+> [!IMPORTANT]
+> The `Worker` was a subclass of <xref:Microsoft.Extensions.Hosting.BackgroundService>. Now, the `TimerService` implements both the <xref:Microsoft.Extensions.Hosting.IHostedService>, and <xref:System.IAsyncDisposable> interfaces.
+
+The `TimerService` is `sealed`, and cascades the `DisposeAsync` call from its `_timer` instance. For more information on the "cascading dispose pattern", see [Implement a `DisposeAsync` method](../../standard/garbage-collection/implementing-disposeasync.md).
+
+When <xref:Microsoft.Extensions.Hosting.IHostedService.StartAsync%2A> is called, the timer is instantiated, thus starting the timer.
+
+> [!TIP]
+> The <xref:System.Threading.Timer> doesn't wait for previous executions of `DoWork` to finish, so the approach shown might not be suitable for every scenario. <xref:System.Threading.Interlocked.Increment%2A?displayProperty=nameWithType> is used to increment the execution counter as an atomic operation, which ensures that multiple threads don't update `_executionCount` concurrently.
+
+Replace the existing `Program` contents with the following C# code:
+
+:::code source="snippets/workers/timer-service/Program.cs" highlight="8":::
+
+The service is registered in `IHostBuilder.ConfigureServices` (*Program.cs*) with the `AddHostedService` extension method. This is the same extension method you use when registering <xref:Microsoft.Extensions.Hosting.BackgroundService> subclasses, as they both implement the <xref:Microsoft.Extensions.Hosting.IHostedService> interface.
```

Comment:
Is the assumption readers are familiar with DI? If not, adding a link to this article would help readers looking for more info.

https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection

Reply:
I suppose, like the previous comment on this - I should just add a "For more info, click DI" link